### PR TITLE
Set up project

### PR DESCRIPTION
**Pull request by martimatix**
This is a PR for the project set up. Work for implementing the business logic will be in the next PR.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "no-param-reassign": 0,
+    "no-unused-vars": ["error", { "args": "none" }],
+    "import/imports-first": ["error", "absolute-first"],
+    "import/prefer-default-export": "off",
+    "import/no-named-as-default": "off",
+    "import/newline-after-import": "error",
+    "import/no-extraneous-dependencies": ["error", {"devDependencies": true}],
+    "prettier/prettier": ["error"]
+  },
+  "parser": "babel-eslint",
```
Comment (dil) [inline .eslintrc.json:18]: doesn't this require babel runtime? I don't see the dependency in package.json

File: serverless.yml [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
 functions:
-  hello:
-    handler: handler.hello
+  preTokenGeneration:
+    handler: src/handler.preTokenGeneration
+    events:
+      - cognitoUserPool:
+          pool: datarock-users-${self:provider.stage}
+          trigger: PreTokenGeneration
+          existing: true
```
Comment (carwinz) [inline serverless.yml:22]: didn't know this was integrated into serverless. Nice!

**Review by carwinz — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+          key: run-tests-v2.1-dependencies-{{ checksum "package-lock.json" }}
+  deploy-test:
+    docker:
+      - image: circleci/node:14
+    steps:
+      - checkout
+      - run:
+          name: Setup test Environment Variables
+          command: |
+            echo export AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID_TEST_ENV >> $BASH_ENV
```
Comment (dil) [inline .circleci/config.yml:38]: Use `Contexts` instead? eg. https://github.com/data-rock/platform-inference-api/blob/master/.circleci/config.yml#L143
Contexts allows us to share the creds between projects.

**Review by dil — APPROVED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (martimatix) [reply .eslintrc.json:18]: Covered by [serverless bundle](https://www.npmjs.com/package/serverless-bundle).

File: serverless.yml (hunk 2 above)
Comment (martimatix) [reply serverless.yml:22]: It's pretty cool. Just discovered that if you manually remove the trigger from the Cognito pool, it won't reattach when redeploying.

**Pull request merged**